### ISSUE TITLE
Incorrect parses XML with comments at the root

### ISSUE THREAD

**Issue by RapsIn4** [ohler55/ox#129]
Example:
https://www.joelosteen.com/Views/RSS/Feed?t=PodcastVideo&ct=CustomList&cst=Podcasts

It only returns the comments when I call:
```ruby
ox_obj = Ox.parse(xml_str)
```

The root is `nil`.

**Comment by ohler55** [comment]
How about just giving me the XML instead of a link to a video.

**Comment by RapsIn4** [comment]
It should have been a link to an xml feed. I copied it to a pastebin since it's quite big.

http://pastebin.com/vgU36i5x

**Comment by ohler55** [comment]
Thanks. That makes is clear what is going on. Ox is a streaming parser. It will parse each XML entity in the file. In your case there are 6 entities in the file. The first being the rss entity and then 5 comments. Ox parses each and return the last entity read. I suppose raising an exception would be better than continuing or allowing for a callback block as Oj does. I'll take a look at that.

In the mean time I'd suggest using the Oj sax parser unless you can wait a few days.

**Comment by RapsIn4** [comment]
Thanks, I'll look into that.

**Comment by ohler55** [comment]
Please try the version I just pushed to github.
``` ruby
Ox.load("<foo><z>x</z></foo><bar>y</bar><!-- what -->") {|x| puts Ox.dump(x) }
```

**Comment by RapsIn4** [comment]
Using `2.3.0a4`

```ruby
Ox.load("<foo><z>x</z></foo><bar>y</bar><!-- what -->").locate('foo');
```

or


```ruby
Ox.parse("<foo><z>x</z></foo><bar>y</bar><!-- what -->").locate('foo');
```

**Returns**
`nil`

**Comment by ohler55** [comment]
I don't think you have grasped the idea of processing multiple XML entities in on file. Please look at the example I gave. The interesting part is not the return value. The return value is only the last entity read. The callback block is the part that is called for each entity read in the file. Your examples do not include a callback block. (you do not need the semicolon at the end of the line either. You must be a C or Java coder. :-)) Try using a callback block as in the example.

**Comment by RapsIn4** [comment]
Thanks. I understand now. I appreciate you walking through this with me.

I think a better example to demonstrate the callback would be:

```ruby
Ox.parse("<foo><z>x</z></foo><bar>y</bar><!-- what -->") {|x| puts x.value }
```

PS: I'm normally very strict on my ruby coding style (my colleagues can attest to that). I've been writing too much Java lately. :stuck_out_tongue:

**Comment by ohler55** [comment]
Great. Let me know if there are any other problems. If not I will ask you to close the issue after the next release.

**Comment by ohler55** [comment]
Just released Ox 2.3.0.

**Comment by RapsIn4** [comment]
:+1:
